src/core/mutatus.scala

```diff
@@ -257,29 +272,39 @@ case class LongId(id: Long) extends IdKey {
 }
 
 /** a data access object for a particular type */
-case class Dao[T](kind: String)(implicit svc: Service, namespace: Namespace) {
+case class Dao[T](kind: String)(
+    implicit svc: Service,
+    namespace: Namespace,
+    decoder: Decoder[T]
+) {
 
   private[mutatus] lazy val keyFactory = {
     val baseFactory = svc.readWrite.newKeyFactory().setKind(kind)
     namespace.option.foldLeft(baseFactory)(_.setNamespace(_))
   }
 
-  /** returns an iterator of all the values of this type stored in the GCP Platform */
-  def all()(implicit decoder: Decoder[T]): Iterator[T] =
-    QueryBuilder(kind).find()(svc, namespace, decoder)
-
-  /** returns query builder */
-  def query(implicit decoder: Decoder[T]): QueryBuilder[T] =
-    QueryBuilder[T](kind)
+  /** returns query builder with empty criteria which fetches all the values of this type stored in the GCP Platform */
+  def all: QueryBuilder[T] = QueryBuilder[T](kind)
```

Comment:
I think we should choose either `all` or `query`, rather than duplicating it. The reason is that someone reading the code may see some code which refers to `all` and other code which refers to `query`. They would wonder what the difference is, when actually there is none. My preference would be for `all`.

Reply:
Should same apply to `apply` and it's alias `run`? In this case `apply` may be enough.

Reply:
Yes, please. Using `apply()` will be more concise, though it won't work if any of the query methods take implicit parameters. I don't think they do, but `run()` is also very short, and it's a very explicit indication to the user about what is in happening in Datastore and what is happening on-heap, so let's use `run()` and drop `apply()`.